packages/@aws-cdk/aws-codebuild/lib/pipeline-actions.ts

```diff
@@ -45,13 +63,23 @@ export class PipelineBuildAction extends codepipeline.BuildAction {
 
     super(parent, name, {
       provider: 'CodeBuild',
+      artifactBounds: { minInputs: 1, maxInputs: 5, minOutputs: 0, maxOutputs: 5 },
       configuration: {
         ProjectName: props.project.projectName,
       },
       ...props,
     });
 
     setCodeBuildNeededPermissions(props.stage, props.project, true);
+
+    handleAdditionalInputOutputArtifacts(props, this,
+      // pass functions to get around protected members
+      (artifact) => this.addInputArtifact(artifact),
+      (artifactName) => this.addOutputArtifact(artifactName));
+  }
+
+  public additionalOutputArtifacts(): codepipeline.Artifact[] {
+    return this._outputArtifacts.slice(1);
```

Comment:
Should we be able to address these by name?

I.e., probably add a function?

```ts
additionalOutputArtifact(name: string): codepipeline.Artifact | Exception
```

Feels like if I want to take one of 5 output artifacts to pass into another step, there should be a convenient way for me to get that object.

(Yes, I know it's just a string wrapper and it feels silly to use a string to get a typed string to pass onto another object that in principle only expected that string THAT YOU ALREADY HAD. So it goes :))

Reply:
If I thought that was silly, I wouldn't be in this business 😀. Added the method!